modules/porous_flow/src/materials/PorousFlowFluidState.C

```diff
@@ -46,7 +46,8 @@ PorousFlowFluidState::PorousFlowFluidState(const InputParameters & parameters)
               : 0),
 
     _num_Z_vars(coupledComponents("z")),
-    _Xnacl(_nodal_material ? coupledDofValues("xnacl") : coupledValue("xnacl")),
+    _is_Xnacl_nodal(isCoupled("xnacl") ? getVar("xnacl", 0)->isNodal() : false),
+    _Xnacl(_nodal_material && _is_Xnacl_nodal ? coupledDofValues("xnacl") : coupledValue("xnacl")),
```

Comment:
coupledValue should work with nodal variables 

what was the trouble?

Reply:
it seems the trouble was coupledDofValues not working with const monomial, not coupledValue not working with nodal

Reply:
Yeah, it causes an assertion or silently makes convergence poor - see #24327 for an example.

Reply:
https://github.com/idaholab/moose/discussions/24327 is with coupledDofValues for const monomial right?

I think we could use coupledValue everywhere. 
But what you have here works

Reply:
The problem (using the above) is that Xnacl could be a nonlinear variable or an aux variable depending on the situation, and that sometimes we need the values at the nodes instead of the qps as some of the kernels are mass-lumped. So we should support nodal nonlinear variables, and both nodal or elemental aux variables here, but calling coupledDofValues on a const monomial aux variable doesn't work (hits an assertion). This seems to be the best work around?

Reply:
this works for me. I didnt know about the mass-lumped